libflux/flux-core/src/semantic/formatter/mod.rs

```diff
@@ -0,0 +1,705 @@
+            //walk::Node::PipeExpr(m) => self.format_pipe_expression(m),
+            walk::Node::ConditionalExpr(m) => self.format_conditional_expression(m),
+            walk::Node::StringExpr(m) => self.format_string_expression(m),
+            walk::Node::ArrayExpr(m) => self.format_array_expression(m),
+            walk::Node::DictExpr(m) => self.format_dict_expression(m),
+            //walk::Node::MemberExpr(m) => self.format_member_expression(m),
+            // walk::Node::UnaryExpr(m) => self.format_unary_expression(m),
+            walk::Node::BinaryExpr(m) => self.format_binary_expression(m),
+            walk::Node::LogicalExpr(m) => self.format_logical_expression(m),
+            // walk::Node::ParenExpr(m) => self.format_paren_expression(m),
+            // walk::Node::FunctionExpr(m) => self.format_function_expression(m),
+            walk::Node::IdentifierExpr(m) => self.format_identifier_expression(m),
+            // walk::Node::Property(m) => self.format_property(m),
+            walk::Node::TextPart(m) => self.format_text_part(m),
+            walk::Node::InterpolatedPart(m) => self.format_interpolated_part(m),
+            walk::Node::StringLit(m) => self.format_string_literal(m),
+            walk::Node::BooleanLit(m) => self.format_boolean_literal(m),
+            walk::Node::FloatLit(m) => self.format_float_literal(m),
+            walk::Node::IntegerLit(m) => self.format_integer_literal(m),
+            walk::Node::UintLit(m) => self.format_unsigned_integer_literal(m),
+            walk::Node::RegexpLit(m) => self.format_regexp_literal(m),
+            walk::Node::DurationLit(m) => self.format_duration_literal(m),
+            walk::Node::DateTimeLit(m) => self.format_date_time_literal(m),
+            // walk::Node::PipeLit(m) => self.format_pipe_literal(m),
+            walk::Node::Identifier(m) => self.format_identifier(m),
+            // walk::Node::ObjectExpr(m) => {
+            //     self.format_record_expression_braces(m, true)
+            // }
+            walk::Node::Package(m) => self.format_package(m),
+            // walk::Node::BuiltinStmt(m) => self.format_builtin(m),
+            _ => self.err = Some(Error::from(format!("bad expression: {:?}", n))),
+        }
+        self.set_indent(curr_ind)
+    }
+
+    fn format_package(&mut self, n: &semantic::nodes::Package) {
+        let pkg_name = &n.package;
+        self.format_package_clause(&semantic::nodes::PackageClause {
+            name: semantic::nodes::Identifier {
+                name: String::from(pkg_name),
+                loc: ast::SourceLocation::default(),
+            },
+            loc: ast::SourceLocation::default(),
+        });
+        for (i, file) in n.files.iter().enumerate() {
+            if i != 0 {
+                self.write_rune('\n');
+                self.write_rune('\n');
+            }
+            self.format_file(file, false)
+        }
+    }
+
+    fn format_monotype(&mut self, n: &MonoType) {
```

Comment:
I was wondering if we might be able to re-use some of the code from the AST formatter. I am not sure how easy that would be... since the underlying structures are different. Any thoughts on this?